tests/integration/mixer/telemetry/metrics/blackhole/blackhole_test.go

```diff
@@ -0,0 +1,117 @@
+//  Copyright 2020 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package blackhole
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/mixer"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/prometheus"
+	"istio.io/istio/pkg/test/framework/components/sleep"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	util "istio.io/istio/tests/integration/mixer"
+)
+
+var (
+	sleepNs    namespace.Instance
+	ist        istio.Instance
+	prom       prometheus.Instance
+)
+
+func TestBlackHoleClusterMetric(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			sleepInst := sleep.DeployOrFail(t, ctx, sleep.Config{Namespace: sleepNs, Cfg: sleep.Sleep})
+			respCode, err := sleepInst.Curl("http://istio.io")
```

Comment:
You can use direct pod ip here of another pod. (Httpbin pod?) we should see same results.

Reply:
+1, strongly recommend not relying on external dependencies. We do this with another test, see https://github.com/istio/istio/blob/ee0f6526b79a5cc6647c512f93158840011cdb6c/tests/integration/pilot/outboundtrafficpolicy/helper.go#L185

Reply:
I don't think relying on another Istio feature i.e. "Sidecar Scoping" is the right way to test external service access. I would recommend using a highly available website (like Google) and test that instead of simulating this behavior via other Istio features.

Creating tests that rely on small number of features and verify the functionality are easier to debug/maintain IMHO.

Reply:
Our tests fail (or more often now, failed, as we have mostly stopped doing it) ALL the time from relying on other "highly available" websites. Relying on external dependencies should be an absolute last resort in a test in my opinion.

Reply:
I see, is there any other way to simulate this other than relying on Istio Sidecar API?

Reply:
I don't understand the concern with relying on Sidecar scoping. Its a first class istio API. But you can also send a request directly to the pod IP address instead of service VIP, but that is even hackier and relying more on implementation details than Sidecar. Not to mention Sidecar is already used in other tests to do this, and has stable for months. we just need to add a metric check to that test

Reply:
The concern is that we need to make sure telemetry works without any networking APIs. Coupling the two in tests does not provide coverage for a use case. It also makes evolution of Sidecar API harder since it'll break this telemetry test.

Reply:
It's scope creep and when someone changes behavior/implementation for Sidecar, there are many unrelated tests that start failing. My difference is just more in philosophy around how tests should be written. I don't think coupling APIs and behaviors in tests is good hygiene.

Reply:
if someone breaks the behavior of sidecar tests SHOULD start failing. I don't think we need some artificial barrier between networking and telemetry. We want totest that blackhole/passthrough works. So why do we need two tests, instead of just one test that makes sure the feature works - which includes telemetry is reported.

Reply:
This split is exactly why networking can accidental break telemetry

Reply:
It's because not everyone is an expert on both Telemetry & Networking APIs and concepts like you @howardjohn.
Coupling concepts for tests increases the barrier for writing, triaging and fixing tests i.e. we are creating more tests which only folks like you can understand and fix.

I'm not going to stop progress here so will defer to more seasoned maintainers decision.

Reply:
See comment https://github.com/istio/istio/pull/22021#issuecomment-599726103